packages/contract/lib/contract/constructorMethods.js

```diff
@@ -154,44 +154,59 @@ module.exports = Contract => ({
     delete this.network.address;
   },
 
+  // accepts 4 input formats
+  //  - (<name>, <address>)
+  //  - (<contractType>) - must have a deployed instance with an address
+  //  - (<contractInstance>)
+  //  - ({ <libName>: <address>, <libName2>: <address2>, ... })
   link(name, address) {
-    // Case: Contract.link(instance)
-    if (typeof name === "function") {
-      const contract = name;
-
-      if (contract.isDeployed() === false) {
-        throw new Error("Cannot link contract without an address.");
-      }
+    switch (typeof name) {
+      case "string":
```

Comment:
Previously the fall-through case allowed anything that _wasn't_ a `"function"` or `"object"` in this case.

Object properties are automatically cast to strings when used to get/set the property. So if someone was using a number, `"true"`, `"false"`, etc as a network name, they previously could use link(1, "address") and it would have worked (i.e., `Object.keys(this.network.links) === ["1"]`).

We probably _shouldn't_ allow this auto-casting behavior, and I actually prefer the way you are doing it in this PR. I just wanted to point out that this _could_ introduce an issue for some users; we should probably have a conversation about it juuussttt in case.

Reply:
Yep, I think that should be considered a bug as well and I agree we shouldn't allow that input.

Reply:
@gnidan, any thoughts on this? :point_up:

Reply:
I think it's fine to get rid of the autocasting